imagersite/imager_images/views.py

```diff
@@ -1,68 +1,151 @@
 
 
-def library_view(request):
-    """Define the library view."""
-    username = request.user.get_username()
+class LibraryView(ListView):
+    """Define the library view class."""
 
-    if username == '':
-        return redirect('home')
-    profile = get_object_or_404(User, username=username)
-    albums = Album.objects.filter(user__username=username)
-    photos = Photo.objects.filter(user__username=username)
+    template_name = 'imager_images/library.html'
+    context_object_name = 'albums_and_photos'
 
-    context = {
-        'profile': profile,
-        'albums': albums,
-        'photos': photos,
-    }
+    def get(self, *args, **kwargs):
+        """Check that user is authenticated, get args and kwargs."""
+        if not self.request.user.is_authenticated:
+            return redirect('home')
 
-    return render(request, 'imager_images/library.html', context)
+        return super().get(*args, **kwargs)
 
+    def get_queryset(self):
+        """Get the context to display."""
+        username = self.request.user.get_username()
+        albums = Album.objects.filter(user__username=username)
+        photos = Photo.objects.filter(user__username=username)
 
-def album_view(request):
-    """Define the library view."""
-    public_albums = Album.objects.filter(published='PUBLIC')
+        return albums, photos
 
-    context = {
-        'public_albums': public_albums,
-    }
+    def get_context_data(self, **kwargs):
+        """Filter the context for display."""
+        context = super().get_context_data(**kwargs)
+        albums = context['albums_and_photos'][0]
+        photos = context['albums_and_photos'][1]
 
-    return render(request, 'imager_images/album.html', context)
+        context.update({
+            'albums': albums,
+            'photos': photos,
+        })
 
+        return context
 
-def photo_view(request):
-    """Define the library view."""
-    public_photos = Photo.objects.filter(published='PUBLIC')
 
-    context = {
-        'public_photos': public_photos,
-    }
+class AlbumView(ListView):
+    """Define the album view class."""
 
-    return render(request, 'imager_images/photo.html', context)
+    template_name = 'imager_images/album.html'
+    model = Album
 
+    def get(self, *args, **kwargs):
+        """Check that user is authenticated, get args and kwargs."""
+        if not self.request.user.is_authenticated:
+            return redirect('home')
 
-def album_detail_view(request, id=None):
-    """Define the library view."""
-    this_album = Album.objects.filter(id=id).first()
+        return super().get(*args, **kwargs)
 
-    context = {
-        'this_album': this_album,
-    }
+    def get_context_data(self, **kwargs):
+        """Filter the context for display."""
+        context = super().get_context_data(**kwargs)
 
-    return render(request, 'imager_images/album_detail.html', context)
+        public_albums = Album.objects.filter(published='PUBLIC')
```

Comment:
Since you are inheriting from the `ListView` class, just specify the `queryset` and the class will return that queryset in the context.